server/main_server/src/main.cpp

```diff
@@ -0,0 +1,1342 @@
+                compiled.validate(_config, decoder);
+                const auto json_result = decoder.get_result();
+
+                if (!json_result.empty()) {
+                    std::ostringstream out;
+                    out << pretty_print(json_result);
+                    fmt::print("{}: {}\n", fn, out.str());
+                    return false;
+                }
+
+                return true;
+            }; // do_validate
+
+            // Validate the server configuration. If that succeeds, move on to validating the
+            // irods_environment.json file.
+            const auto schema_dir = irods::get_irods_home_directory() / fmt::format("configuration_schemas/v{}", IRODS_CONFIGURATION_SCHEMA_VERSION);
+            const auto server_config_schema = schema_dir / "server_config.json";
+            if (do_validate(config, server_config_schema.c_str())) {
+                std::string env_file;
+                std::string session_file;
+                if (const auto err = irods::get_json_environment_file(env_file, session_file); !err.ok()) {
+                    fmt::print("{}: {}\n", __func__, err.status());
+                    return false;
+                }
+
+                // Validate the irods_environment.json file referenced by the server configuration.
+                std::ifstream in{env_file};
+                if (!in) {
+                    return false;
+                }
+                const auto env_file_config = jsoncons::json::parse(in);
+                const auto svc_acct_schema = schema_dir / "service_account_environment.json";
+                return do_validate(env_file_config, svc_acct_schema.c_str());
+            }
+        }
+        catch (const std::exception& e) {
+            fmt::print("{}: {}\n", __func__, e.what());
+        }
+
+        return false;
+    } // validate_configuration
+
+    auto daemonize() -> void
+    {
+        // Become a background process.
+        switch (fork()) {
+            case -1: _exit(1);
+            case  0: break;
+            default: _exit(0);
+        }
+
+        // Become session leader.
+        if (setsid() == -1) {
+            _exit(1);
+        }
+
+        // Make sure we aren't the session leader.
+        switch (fork()) {
+            case -1: _exit(1);
+            case  0: break;
+            default: _exit(0);
+        }
+
+        umask(0);
+
+        // Change the working directory to the root directory. Using the root directory is
+        // common and protects the OS by allowing it to unmount file systems. If the working
+        // directory of the daemon existed on a file system other than the one containing "/",
+        // the OS would not be able to unmount it.
+        //
+        // It's also possible to change the working directory to other directories, for example,
+        // the iRODS home directory. However, that directory normally exists under /var.
+        chdir("/");
```

Comment:
This means that everything involving the local filesystem MUST use an absolute path. We have the default paths functions for this.

Any thoughts on this?

Reply:
This is fine.